synapse/lib/filepath.py

```diff
@@ -500,9 +656,8 @@ def isfile(*paths):
 
     if not fpath:
         return False
-    if fpath.isfile():
-        return True
-    return False
+    fpath.close()
```

Comment:
There may be spooled temp files still open.  Try running the tests with that line commented out ;)